Solved-Show default collation when creating new DB #12984

nfebe opened the pull request:
#12984

ibennetch commented:
Something may have gone wrong, because there are 2,268 files changed with this pull request.

Some of it looks like permission problems, but often this can be caused by working from one branch (such as QA_4_7) and creating a pull request against a different branch (master, for instance).

nijel committed:
# This is a combination of 11 commits.
# The first commit's message is:
# This is a combination of 3 commits.
# The first commit's message is:
Remove debugging code

Fixes #13059

Signed-off-by: Michal Čihař <michal@cihar.com>

# This is the 2nd commit message:

added default value for collation dropdown

# This is the 3rd commit message:

Translated using Weblate (Chinese (Traditional))

Currently translated at 99.9% (3191 of 3194 strings)

[CI skip]

# This is the 2nd commit message:

Remove documentation for old MySQL server

We're supporting 5.5 and newer, while this has been changed in 5.1.

Issue #13062

Signed-off-by: Michal Čihař <michal@cihar.com>

# This is the 3rd commit message:

updated method for getting default collation

# This is the 4th commit message:

removed trailing white spaces

# This is the 5th commit message:

updated with upstream and signed commits

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

# This is the 6th commit message:

signed off commit

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

# This is the 7th commit message:

removed trailing characters

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

# This is the 8th commit message:

removed unused import

# This is the 9th commit message:

Made default collation changes more specific to the required page

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

# This is the 10th commit message:

fixed trailing white spaces in two unrelated files

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

# This is the 11th commit message:

updating last commit

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

Fenn-CS commented:
@ibennetch  they are permission problems I work on ubuntu I had to change the permission to 777 to have full access on my apache server...How do I fix this pull request?

nfebe committed:
added default value for collation dropdown

Fenn-CS commented:
@ibennetch I have fixed every issue and only the file that needs to be changed to fix the error is marked changed now. Thanks

Fenn-CS commented:
![changes](https://cloud.githubusercontent.com/assets/14317775/23761421/05eef5a8-04f3-11e7-8d0d-b119d48cae29.png)

OlafvdSpek commented:
The whitespace changes shouldn't be there.

ibennetch commented:
Indeed, the whitespace changes make it difficult to see what this accomplishes (even though, it seems the switch/case indentation could be fixed, that should be a different pull request). 

Perhaps your editor was making whitespace "fixes" without your knowledge; many editors that do autoindentation can do that inadvertently.

Also, your commit is missing the "signed off by" comment that indicates you agree with and abide by the terms of the "Developer Certificate of Origin"; more information about this is linked by phpmyadmin-bot above.

ibennetch commented:
I just tested this by changing the default collation of my server and the displayed value did not update; I'm not sure from where exactly you're gathering that information but it may not be correct.

Fenn-CS commented:
@ibennetch does this mean I supposed to close this pull request in order to fix the issue and then create a new pr?

ibennetch commented:
No, you can keep this open and push more commits to your branch. When you do so, the pull request automatically updates.

I prefer that because it's easier and less noisy than opening a new pull request.

Fenn-CS commented:
@ibennetch ok thanks, on the test that failed this is how I get the information about default collation 
`
$sql='SELECT DEFAULT_CHARACTER_SET_NAME, DEFAULT_COLLATION_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME ='.'"information_schema"'`;

nfebe committed:
Merge https://github.com/phpmyadmin/phpmyadmin

Fenn-CS commented:
It simply gets the information from the information_schema table. Maybe you check whether actually the default collation was actually changed. I think I have been able to fix the indentation problems (white space changes) with the pull request also thanks.

ibennetch commented:
If I'm not mistaken, that only tells you the default collation for the information_schema database, not for the whole system.

devenbansod commented on libraries/Charsets.php:
```diff
@@ -191,37 +191,46 @@ public static function getCharsetDropdownBox(
         if ($label) {
+         //Find default collation value
+            $sql = 'SELECT DEFAULT_CHARACTER_SET_NAME, DEFAULT_COLLATION_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME ='.'"information_schema"';
+            $res = $GLOBALS['dbi']->query($sql);
```
@Fenn-CS I suggest you use the existing function `getServerCollation()` defined in `libraries/DatabaseInterface.php` for getting the server collation (which will act as a default for a newly created database, if nothing is specified), instead of separately querying it using a raw query (which is not returning the correct value that we should look at actually) in the code here.

devenbansod commented:
@ibennetch yes. That's right. 

Also, @Fenn-CS I can still see the whitespace changes in this PR, please remove them while committing with the requested changes.

nfebe committed:
updated method for getting default collation

nfebe committed:
removed trailing white spaces

nfebe committed:
updated with upstream and signed commits

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

nfebe committed:
signed off commit

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

Fenn-CS commented:
I am using sublime text and am having a hard time with these white space changes... Any help will be appreciated thanks.

nfebe committed:
Merge branch 'master' of https://github.com/Fenn-CS/phpmyadmin

nfebe committed:
removed trailing characters

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

Fenn-CS commented:
@devenbansod I have applied the changes your requested. Thnks

devenbansod commented on libraries/Charsets.php:
```diff
@@ -7,6 +7,7 @@
 namespace PMA\libraries;
 use PMA\libraries\Util;
+use PMA\libraries\DatabaseInterface;
 
```
Unused import ?

devenbansod commented on libraries/Charsets.php:
```diff
@@ -45,7 +46,7 @@ class Charsets
         'windows-1257' => 'cp1257',
-    );
+        );
 
```
It seems that the whitespace problem still persists. Please fix this.

devenbansod commented on libraries/Charsets.php:
```diff
@@ -145,32 +146,32 @@ public static function getMySQLCollationsDefault()
         $submitOnChange = false
-    ) {
+        ) {
         self::loadCharsets();
```
I believe that this is not the correct place to make this change. 
The function `getCharsetDropdownBox` already takes a default value as a parameter `$default` and we don't every charset dropdown to have a default value as the server collation.

For example, the table creation might use the database collation as the default value and not the server collation.

Fenn-CS commented:
@devenbansod I am having challenges fixing the white space issues...They seem to be created by sublime text how do I combat this?

nfebe committed:
removed unused import

devenbansod commented:
@Fenn-CS I am not sure what is the problem, since even I use Sublime text sometimes, but it does not create any issue at all. It might be some configuration issue, so you can better ask for a solution to your problem on their support forum or on Stackoverflow. Till then you may want to use a different text editor. 

Also, use `git commit -s` while commiting, to avoid the error regarding the missing Signed-Off-By line.

nfebe committed:
Made default collation changes more specific to the required page

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

Fenn-CS commented:
@devenbansod I have moved the changes from the general function to the specific page where the performance was expected now the default collation value is passed like a parameter to the function I previously edited. I have also addressed the issue with Trailing white spaces as from my last commit. Thanks

devenbansod commented on libraries/Charsets.php:
```diff
@@ -472,4 +472,4 @@ public static function getCollationDescr($collation)
         return $descr;
     }
-}
+}
```
There are just two extra space changes in unrelated files. Please remove them and then [squash](https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History#_squashing) all the commits into one.

nfebe committed:
fixed trailing white spaces in two unrelated files

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

nfebe committed:
updating last commit

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

nfebe committed:
canceled white space changes in two unrelated files by updating from upstream

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

nfebe committed:
Merge branch 'master' of https://github.com/Fenn-CS/phpmyadmin

nfebe committed:
reverted changes edited files not concerned

Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>

Fenn-CS commented:
Hello @devenbansod and @ibennetch I have made all the changes requested. I think every thing is ok now apart from the fact I have been unable to squash all the commits into one. I will need help on that I tried doing that and it seemed to revert my work bringing back old changes. Apart from that I think it solves the  problem as required right now...

devenbansod commented:
I am squashing and merging your commits.

Also a suggestion regarding your next PRs, please make a new branch while implementing a feature/fixing a bug, instead of directly working on master/QA_* branches on your fork.

Thanks for your contribution!

Fenn-CS commented:
@devenbansod thanks. My pleasure to work with phpMyAdmin! :)
